src/components/Detailpage/Detailpage.js

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { ReactComponent as Gluten } from '../../images/icons/gluten.svg'
+import { ReactComponent as Soy } from '../../images/icons/soy.svg'
+import { ReactComponent as Arrow } from '../../images/icons/back-arrow.svg'
+import { useHistory, useParams } from 'react-router-dom'
+
+export default function Detailpage({ category }) {
+  let { id } = useParams()
+  useEffect(() => {
+    fetchDetail()
+  }, [])
+
+  const [detail, setDetail] = useState({
+    allergies: [],
+  })
+
+  const fetchDetail = async () => {
+    const fetchDetail = await fetch(`http://localhost:4000/${category}/${id}`)
+    const detail = await fetchDetail.json()
+    setDetail(detail)
+  }
+
+  let history = useHistory()
+
+  function handleClick() {
+    history.goBack()
+  }
+
+  return (
+    <>
+      <ImageSection>
+        <ArrowStyled onClick={handleClick} />
+        <ProductName>{detail.title}</ProductName>
```

Comment:
Please consider destructuring `detail` right before the `return` statement, so you have `title, image, ingredients` and `detail` as consts.